test/RampageRequestHandlerTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace Horde\Http\Server\Test;
+use \PHPUnit\Framework\TestCase;
+use Horde\Http\RequestFactory;
+use Horde\Http\ResponseFactory;
+use Horde\Http\StreamFactory;
+use Horde\Http\Server\RampageRequestHandler;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class RampageRequestHandlerTest extends TestCase
+{
+    public function testAddMiddleware()
+    {
+        $responseFactory = new ResponseFactory;
+        $streamFactory = new StreamFactory;
+        $handler = new RampageRequestHandler($responseFactory, $streamFactory);
+        $middlewareMock1 = $this->createMock(MiddlewareInterface::class);
+        $middlewareMock1->n = 1;
+        $middlewareMock2 = $this->createMock(MiddlewareInterface::class);
+        $middlewareMock2->n = 2;
+        $middlewareMock3 = $this->createMock(MiddlewareInterface::class);
+        $middlewareMock3->n = 3;
+        $handler->addMiddleware($middlewareMock1);
+        $handler->addMiddleware($middlewareMock2);
+        $handler->addMiddleware($middlewareMock3);
+        $firstInNumber = $handler->nextMiddleware()->n;
```

Comment:
Check the other middlewares as well. Also test if it returns `null` after the middleware stack has been exhausted.